sim/paladin/righteous_fury.go

```diff
@@ -0,0 +1,71 @@
+package paladin
+
+import (
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+func (paladin *Paladin) registerRighteousFury() {
+	if !paladin.Options.RighteousFury {
+		return
+	}
+	horRune := proto.PaladinRune_RuneHandsHandOfReckoning
+	hasHoR := paladin.hasRune(horRune)
+	actionID := core.ActionID{SpellID: core.TernaryInt32(hasHoR, int32(horRune), 25780)}
+
+	rfThreatMultiplier := 0.6 + core.TernaryFloat64(hasHoR, 0.2, 0.0)
+	// Improved Righteous Fury is multiplicative.
+	rfThreatMultiplier *= 1.0 + []float64{0.0, 0.16, 0.33, 0.5}[paladin.Talents.ImprovedRighteousFury]
+
+	paladin.OnSpellRegistered(func(spell *core.Spell) {
+		if spell.SpellSchool.Matches(core.SpellSchoolHoly) {
+			spell.ThreatMultiplier *= 1.0 + rfThreatMultiplier
+		}
+	})
+
+	if !hasHoR { // This is just a visual/UI indicator when we don't have HoR rune.
```

Comment:
What do you think about instead of using an if/else here, you can go ahead and create the righteous fury aura with one setup. Then if you don't have the HoR rune just return. Anything past that means you have HoR, so do the rest of the setup there.

```
paladin.RighteousFuryAura = core.MakePermanent(paladin.RegisterAura(core.Aura{
  ActionID: actionID
  Label: "Righteous Fury",
}))

if !hasHoR {
  return
}

// Passive effects granted by Hand of Reckoning rune.

// Damage which takes you below 35% health is reduced by 20% (DR component of WotLK's Ardent Defender)
rfDamageReduction := 0.2

		handler := func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
			incomingDamage := result.Damage
			if (paladin.CurrentHealth()-incomingDamage)/paladin.MaxHealth() <= 0.35 {
				result.Damage -= (paladin.MaxHealth()*0.35 - (paladin.CurrentHealth() - incomingDamage)) * rfDamageReduction
				if sim.Log != nil {
					paladin.Log(sim, "Righteous Fury absorbs %d damage", int32(incomingDamage-result.Damage))
				}
			}
		}

		paladin.AddDynamicDamageTakenModifier(handler)

paladin.RighteousFuryAura.OnHealTaken = func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
				if spell.IsOtherAction(proto.OtherAction_OtherActionHealingModel) {
					manaGained := result.Damage * 0.25
					paladin.AddMana(sim, manaGained, horManaMetrics)
				}
			}

```

sorry for the formatting it's really annoying copying/pasting and dealing with spacing in GitHub 😅 let me know if this makes sense